Sync v2-7-stable with v2-7-test to release 2.7.0rc1 (#33247)
mistercrunch/Airflow: RELEASE_NOTES.rst

Improvements
""""""""""""
- Configurable health check threshold for triggerer (#33089, #33084)

> - Add documentation generation for CLI commands from executors (#33081, #33143, #33081)

Reviewer: I think this can be in misc section?

Author: Just checked now, and it's actually part of AIP-51 but not marked so, and I didn't also include AIP-51 in this changelog, I wonder which other one I'm missing? I will skip this and mark them all as part of AIP-51

Docs only changes
"""""""""""""""""
- Add links to ``DAGRun / DAG / Task`` in templates-ref.rst (#33013)

> - Add example for list-import-errors as a CI check (#32811)

Reviewer: ```suggestion
- Add docs of how to test for DAG Import Errors (#32811)
```

I think this explains better to users what was added
